qcodes/instrument_drivers/Andor/DU401.py

```diff
@@ -0,0 +1,424 @@
+
+    def set_temperature(self, temperature):
+        c_temperature = ctypes.c_int(temperature)
+        code = self.dll.SetTemperature(c_temperature)
+        self.error_check(code, 'SetTemperature')
+
+    def set_trigger_mode(self, mode):
+        c_mode = ctypes.c_int(mode)
+        code = self.dll.SetTriggerMode(c_mode)
+        self.error_check(code, 'SetTriggerMode')
+
+    def shut_down(self):
+        code = self.dll.ShutDown()
+        self.error_check(code, 'ShutDown')
+
+    def start_acquisition(self):
+        code = self.dll.StartAcquisition()
+        self.error_check(code, 'StartAcquisition')
+
+    def wait_for_acquisition(self):
+        code = self.dll.WaitForAcquisition()
+        self.error_check(code, 'WaitForAcquisition')
+
+
+class Spectrum(Parameter):
+    """
+    Parameter class for a spectrum taken with an Andor CCD.
+
+    The spectrum is saved in a list with the length being set by the number of pixels on the CCD.
+
+    Args:
+        name (str): Parameter name.
+
+    """
+
+    def __init__(self, name, *args, **kwargs):
+
+        self.ccd = kwargs['instrument']
+        super().__init__(name, *args, **kwargs)
+
+    def get_raw(self):
+
+        # get acquisition mode
+        acquisition_mode = self.ccd.acquisition_mode.get()
+
+        # start acquisition
+        self.ccd.atmcd64d.start_acquisition()
+
+        # wait for single acquisition
+        if acquisition_mode == 'single scan':
+            self.ccd.atmcd64d.wait_for_acquisition()
+
+        # wait for accumulate acquisition
+        elif acquisition_mode == 'accumulate':
+            number_accumulations = self.ccd.number_accumulations.get()
+            for i in range(number_accumulations):
+                self.ccd.atmcd64d.wait_for_acquisition()
+
+        # get and return spectrum
+        return self.ccd.atmcd64d.get_acquired_data(self.ccd.x_pixels)
+
+
+class Andor_DU401(Instrument):
+    """
+    Instrument driver for the Andor DU401 BU2 CCD.
+
+    Args:
+        name (str): Instrument name.
+        dll_path (str): Path to the atmcd64.dll file. If not set, a default path is used.
+        camera_id (int): ID for the desired CCD.
+        setup (bool): Flag for the setup of the CCD. If true, some default settings will be sent to the CCD.
+
+    Attributes:
+        serial_number (int): Serial number of the CCD.
+        head_model (str): Head model of the CCD.
+        firmware_version (int): Firmware version of the CCD.
+        firmware_build (int): Firmware build of the CCD.
+        x_pixels (int): Number of pixels on the x axis.
+        y_pixels (int): Number of pixels on the y axis.
+
+    """
+
+    # TODO (SvenBo90): implement further acquisition modes
+    # TODO (SvenBo90): implement further read modes
+    # TODO (SvenBo90): implement further trigger modes
+    # TODO (SvenBo90): add and delete parameters dynamically when switching acquisition mode or read mode
+    # TODO (SvenBo90): handle shutter closing and opening timings
```

Comment:
these are worth mentioning in the docstring of the driver a format similar to "Note that the driver does not support this and that". Perhaps, also in the example notebook.